src/Compilers/CSharp/Portable/Symbols/NativeIntegerTypeSymbol.cs

```diff
@@ -93,50 +162,241 @@ internal NativeIntegerTypeSymbol(NamedTypeSymbol underlyingType) : base(underlyi
 
         internal sealed override NamedTypeSymbol NativeIntegerUnderlyingType => _underlyingType;
 
-        internal override bool Equals(TypeSymbol t2, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null) => _underlyingType.Equals(t2, comparison, isValueTypeOverrideOpt);
+        internal override bool Equals(TypeSymbol? other, TypeCompareKind comparison, IReadOnlyDictionary<TypeParameterSymbol, bool>? isValueTypeOverrideOpt = null)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+            if ((object)this == other)
+            {
+                return true;
+            }
+            if (!_underlyingType.Equals(other, comparison, isValueTypeOverrideOpt))
```

Comment:
>_underlyingType [](start = 17, length = 15)

By going to the underlying type, aren't we ignoring native integers?  If so, it doesn't appear that we've checked that the flag permits doing so.